src/Interpreters/HashJoin.cpp

```diff
@@ -1990,10 +2056,20 @@ void HashJoin::reuseJoinedData(const HashJoin & join)
     }
 }
 
-BlocksList HashJoin::releaseJoinedBlocks()
+BlocksList HashJoin::releaseJoinedBlocks(bool restructure)
 {
+    LOG_DEBUG(log, "Join data is being released, {} bytes and {} rows in hash table", getTotalByteCount(), getTotalRowCount());
```

Comment:
Probably LOG_TRACE? Doesn't it look like candidate for ProfileEvent, as well as other logs related to memory/rows consumption?

Reply:
Indeed, I'll check what we already have and how this ones will fit.